ops/_main.py

```diff
@@ -523,15 +525,43 @@ def _commit(self):
         """Commit the framework and gracefully teardown."""
         self.framework.commit()
 
+    def _eval(self, expr: str):
+        """Eval an expression in the context of the charm and print the result to stdout."""
+        import json
+
+        globs = {'self': self.charm, 'ops': ops, 'json': json}
```

Comment:
I feel a slippery slope here: start with self, add ops, then `json.dumps()` because surely it's useful, then maybe yaml? pathlib? os? stdlib operator?

Reply:
Yeah, I agree. `self` seems fine as you kinda need the charm instance, `ops` seems reasonable, but anything beyond that I'd see if we can find a different way to import stuff. If we go the "exec" route (rather than "eval"), I guess you can do `import json; json.dumps(x)`.

Reply:
to be fair, jhack also has a sister command, `jhack script`, that allows loading in a python file with a `main(charm:Charm)` entrypoint for the kind of 'more complex' scenarios where you may need to import stuff and the like

Reply:
eval is mostly for inspection, not so much mutation. json felt like a universal, low-barrier addition. Yaml I wouldn't object to, but pathlib, os, operator feel too much. It's not about allowing you to golf complex operations into a string, it's about inspecting live data